qa/common/infer_util.py

```diff
@@ -278,18 +248,67 @@ def infer_exact(tester,
         input1_byte_size = sum([i1.nbytes for i1 in input1_list_tmp])
 
-    # Create system/cuda shared memory regions if needed
-    shm_regions, shm_handles = su.create_set_shm_regions(
-        input0_list_tmp, input1_list_tmp, output0_byte_size, output1_byte_size,
-        outputs, shm_region_names, precreated_shm_regions,
-        use_system_shared_memory, use_cuda_shared_memory)
-
     if model_version is not None:
         model_version = str(model_version)
     else:
         model_version = ""
 
     # Run inference and check results for each config
+    inferAndCheckResults(tester, configs, pf, batch_size, model_version,
+                         input_dtype, output0_dtype, output1_dtype,
+                         tensor_shape, input0_array, input1_array,
+                         output0_array, output1_array, output0_raw, output1_raw,
+                         outputs, precreated_shm_regions, input0_list_tmp,
+                         input1_list_tmp, shm_region_names, input0_byte_size,
+                         input1_byte_size, output0_byte_size, output1_byte_size,
+                         use_system_shared_memory, use_cuda_shared_memory,
+                         skip_request_id_check)
+
+
+def inferAndCheckResults(tester, configs, pf, batch_size, model_version,
+                         input_dtype, output0_dtype, output1_dtype,
+                         tensor_shape, input0_array, input1_array,
+                         output0_array, output1_array, output0_raw, output1_raw,
+                         outputs, precreated_shm_regions, input0_list_tmp,
+                         input1_list_tmp, shm_region_names, input0_byte_size,
+                         input1_byte_size, output0_byte_size, output1_byte_size,
+                         use_system_shared_memory, use_cuda_shared_memory,
+                         skip_request_id_check):
+    # Lazy shm imports...
+    if use_system_shared_memory or use_cuda_shared_memory:
+        import tritonclient.utils.shared_memory as shm
+        import tritonclient.utils.cuda_shared_memory as cudashm
+    num_classes = 3
+
+    # Get model platform
+    model_name = tu.get_model_name(pf, input_dtype, output0_dtype,
+                                   output1_dtype)
+    if configs[0][1] == "http":
+        metadata_client = httpclient.InferenceServerClient(configs[0][0],
+                                                           verbose=True)
+        metadata = metadata_client.get_model_metadata(model_name)
+        platform = metadata["platform"]
+    else:
+        metadata_client = grpcclient.InferenceServerClient(configs[0][0],
+                                                           verbose=True)
+        metadata = metadata_client.get_model_metadata(model_name)
+        platform = metadata.platform
+
+    INPUT0 = "INPUT0"
+    INPUT1 = "INPUT1"
+
+    if platform == "pytorch_libtorch":
+        OUTPUT0 = "OUTPUT__0"
+        OUTPUT1 = "OUTPUT__1"
```

Comment:
FYI, we probably can drop the special handling for libtorch? @Tabrizian

Reply:
Yes, I think we can drop this now that we support   STRICT_CONFIG_ORDERING naming convention.

Reply:
This requires a change in the model gen script https://github.com/triton-inference-server/server/blob/main/qa/common/gen_qa_models.py#L1389-L1445. I would rather change this in another PR